File: Go/Solutions/projeuler/Problem010.go
package euler

import (
	"fmt"
	"time"
)

/*
Problem010 answers the problem at : https://projecteuler.net/problem=10
	* Problem 10:
		Find the sum of all the primes below two million.
*/
func Problem010() {

	start := time.Now()
	ans := Sum(PrimeSieve(2000000))
	end := time.Now()

	fmt.Printf("\nAnswer to Problem 10 : %d\n", ans)
	fmt.Printf("Time Taken: %f seconds\n\n", end.Sub(start).Seconds())
}

//Sum reports the sum of all elements of [arr]
func Sum(arr []int) (s int) {
	for _, i := range arr {
		s += i
	}
	return
}

File: Python/Problem010.py
from math import sqrt; from itertools import count, islice
from time import time

def prime(n):
    return n > 1 and all(n%i for i in islice(count(2), int(sqrt(n)-1)))

start = time()
ans = sum(filter(lambda x: prime(x), range(2, 2*10**6)))

if __name__ == '__main__':
    start = time()
    print(f"\nAnswer: { ans }")
    print(f"Time Taken: { time() - start }\n")
